### PR TITLE
Adds a mind-numbingly simple SHA2 PRNG.

### PR DESCRIPTION
**Pull request by cubicool**
This patch adds a trivial counter/SHA2 hash PRNG generator so simple and so
boring as to be borderline offensive.

### CODE REVIEW

File: lib/generators/sha512.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+require 'digest/sha2'
+
+class SHA512Generator < Generator
+  def initialize(*args)
+    super(*args)
+    @i = 0
+  end
+
+  def self.summary
+    "SHA2 hash with a 512-byte bit length"
```
Comment (kkinion) [inline lib/generators/sha512.rb]: This description needs a minor update: `512-bit length`

**Comment by kkinion** [comment]
Thank you for correcting the summary of sha512.

`devrand` is neat idea; however, there are two problems with this approach:
- it is not deterministic, so it should not be added as a generator
- `/dev/random` blocks indefinitely, making this impossible to run with with the entrospection tool

**Comment by cubicool** [comment]
On 04/24/2015 03:28 PM, kkinion wrote:

> Thank you for correcting the summary of sha512.
> 
> |devrand| is neat idea; however, there are two problems with this 
> approach:
> - it is not deterministic, so it should not be added as a generator

Well, that was the joke. :) Maybe it IS deterministic, if the rumors are 
to be believed. :)

> - |/dev/random| blocks indefinitely, making this impossible to run
>   with with the entrospection tool

I think it only blocks until the kernel decides some entropy threshold 
has been met, which can seem like a long time. The manpage says:

"When the entropy pool is empty, reads from /dev/random will block until 
additional environmental noise is gathered."

Anyways, the real point of the commit was to get a chuckle from the 
Intel joke. :) Plus, I've never coded in Ruby (beyond using Puppet), and 
I've been exploring it the last 24 hours or so...

**Comment by kkinion** [comment]
Ah, ok, I get the joke. Well, I'll merge this once you've removed the non-deterministic generator (or find a backdoor that works on all *nix systems).

**Pull request merged**